pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -276,6 +284,10 @@ func parseBaseOS(getconfigCtx *getconfigContext,
 		RetryUpdateCounter: getconfigCtx.configRetryUpdateCounter,
 		Activate:           baseOS.Activate,
 	}
+	// If Activate is set, we need to activate the new baseOS
+	if cfg.Activate {
+		activateNewBaseOSFlag = true
```

Comment:
This doesn't capture the change i.e., where zedagent has already published the same BaseOsConfig with activate = false and now it is publishing it as true.

The loop below using GetAll to find the ones to delete can be used to also check if Activate went from false to true on any of the existing BaseOsConfig structs or if this is a brand new Key and it has Activate true.

Reply:
Ah, that makes sense; I've missed this case! Thanks for pointing this out; I'll adapt the code now.

Reply:
Hmm, I think we cannot use the loop below to detect changes in the Activate field, as this loop is reached only when a new BaseOS config has arrived.
So, I think. this check should be performed earlier, even if the BaseOS config is not present in this version of the config at all.

Reply:
The loop below runs any time the above sha check indicates there there is some change.